### DISCUSSION TITLE
How to rebind up and down keys?

### DISCUSSION THREAD

**Discussion by mashtonian** [d99kris/nchat#123] [Q&A]
I changed the key bindings for up and down to the commonly used Ctrl-P and Ctrl-N. This makes the address book a lot easier to navigate for me, with my hands on the home keys.

Unfortunately this then broke the ability to navigate up and down through messages in the conversation window, using either the up / down arrow keys or Ctrl N / P.

It is also interesting that having made this mapping the Ctrl-N key retained the ability to open the address book, as well as scrolling downward within it, once opened.

Ideally, I would like to be able to use Ctrl-N/P for up and down within the address book, whilst retaining Ctrl-N to open it, as well as the use of arrow keys for selecting messages in a conversation. Conceptually, a way of doing this would be to allow multiple keys to be mapped to the "up" and "down" actions, perhaps?

Is this possible, am I missing something?

**Comment by d99kris** [comment]
Hi @mashtonian - `nchat` does not implement a "modal" user interface currently, so message selection and opening address book cannot be the same key binding.

Can you perhaps assign a different key binding for opening the address book, like `alt-a` or something?

You can run `nchat -k` to run its "key code dump mode" and press `alt-a` to determine the key code to use in the conf file.
`q` or `ctrl-c` exits the key dump mode.
